src/Sparrow/Json/Parsing/JsonParserState.cs

```diff
@@ -193,19 +209,21 @@ public static void FindEscapePositionsIn(FastList<int> buffer, byte* str, ref in
                 }
             }
         }
+
         private static void ThrowInvalidSizeForEscapeControlChars(int previousComputedMaxSize)
         {
             throw new InvalidOperationException($"The previousComputedMaxSize: {previousComputedMaxSize} is too small to support the required escape positions. Did you not call FindMaxNumberOfEscapePositions?");
         }
+
         public int WriteEscapePositionsTo(byte* buffer)
         {
             var escapePositions = EscapePositions;
             var originalBuffer = buffer;
             WriteVariableSizeInt(ref buffer, escapePositions.Count);
 
-            // PERF: Using a for in this way will evict the bounds-check and also avoid the cost of using an struct enumerator. 
-            for (int i = 0; i < escapePositions.Count; i++)
-                WriteVariableSizeInt(ref buffer, escapePositions[i]);
+            // PERF: Using a for-each in this way will evict the bounds-check and also avoid the cost of using a struct enumerator. 
+            foreach (var escapePosition in escapePositions)
```

Comment:
So previously we preferred `for` and now `foreach` does better, right? What makes is so special here (since we don't make such change in other places)?

Reply:
This is a new optimization, it only really make sense where you are dealing with very hot code (like this)... for the most part, it is just a micro-optimization.

Reply:
However, the old optimization should still apply, no?
I would expect both to generate the same machine code, no?

Reply:
If you are lucky this idiom will also trigger it, but in this case, the code is even better; so that's why I changed it.
```
            for (int i = 0; i < escapePositions.Count; i++)
                WriteVariableSizeInt(ref buffer, escapePositions[i]);
```